### PR TITLE
Topic/imx/sai fix 4 playback

### PR DESCRIPTION
**Pull request by groncarolo**
This patch set corrects:
- topology link name
- clock divider in SAI dai
- configures frame_bytes from format
:x

### CODE REVIEW

File: src/audio/dai.c [hunk 1]
```diff
@@ -798,6 +798,19 @@ static int dai_config(struct comp_dev *dev, struct sof_ipc_dai_config *config)
...
+			dd->frame_bytes = 2;
+			break;
+		case SOF_IPC_FRAME_S24_4LE:
+		case SOF_IPC_FRAME_S32_LE:
+			dd->frame_bytes = 4;
+			break;
+		default:
+			return -EINVAL;
+		}
+
```
Comment (paulstelian97) [inline src/audio/dai.c]: It did look weird to me that these were missing indeed

File: tools/topology/sof-imx8qxp-wm8960.m4 [hunk 2]
```diff
@@ -60,9 +60,9 @@ dnl PCM_PLAYBACK_ADD(name, pcm_id, playback)
...
 
 dnl DAI_CONFIG(type, idx, link_id, name, sai_config)
-DAI_CONFIG(SAI, 1, 0, SAI1-Codec,
-	SAI_CONFIG(I2S, SAI_CLOCK(mclk, 49152000, codec_mclk_in),
-		SAI_CLOCK(bclk, 3072000, codec_slave),
-		SAI_CLOCK(fsync, 48000, codec_slave),
+DAI_CONFIG(SAI, 1, 0, be.wm8960-hifi,
+	SAI_CONFIG(I2S, SAI_CLOCK(mclk, 12288000, codec_mclk_in),
+		SAI_CLOCK(bclk, 3072000, codec_master),
+		SAI_CLOCK(fsync, 48000, codec_master),
```
Comment (paulstelian97) [inline tools/topology/sof-imx8qxp-wm8960.m4]: I presume these are the correct values (does it work?)

File: src/audio/dai.c (hunk 1 above)
Comment (groncarolo) [reply src/audio/dai.c]: We had the patch but did not push it

File: tools/topology/sof-imx8qxp-wm8960.m4 (hunk 2 above)
Comment (groncarolo) [reply tools/topology/sof-imx8qxp-wm8960.m4]: It does

**Comment by jajanusz** [comment]
@groncarolo Please set up your GitHub profile correctly - as you can see GH didn't link your commits to your profile - probably because it couldn't match email/user name with ones in the GH profile page.

**Comment by paulstelian97** [comment]
> @groncarolo Please set up your GitHub profile correctly - as you can see GH didn't link your commits to your profile - probably because it couldn't match email/user name with ones in the GH profile page.

I believe the commits were linked fine even before, it's just his profile picture that might be a bit confusing.

**Comment by groncarolo** [comment]
@jajanusz Thanks I added my work email hope it works

@dbaluta  added formula for DIV

File: src/include/sof/drivers/sai.h [hunk 3]
```diff
@@ -225,7 +225,11 @@
 #define SAI_FLAG_PMQOS   BIT(0)
 
 #define SAI_FIFO_WORD_SIZE	64
-#define SAI_CLOCK_DIV		0x3
+/* Divides down the audio master clock to generate the bit clock when
+ * configured for an internal bit clock.
+ * The division value is (DIV + 1) * 2.
+ */
```
Comment (paulstelian97) [inline src/include/sof/drivers/sai.h]: I believe mentioning why you need to divide by 16 (if I understood this explanation correctly) might be nice, with the hardcoded params.

Comment (groncarolo) [reply src/include/sof/drivers/sai.h]: This define is used by @jlacla commit fb068e6a7a1670d3f15032f27651fbe70851bccb
for now as in future will be determined dinamically

Comment (paulstelian97) [reply src/include/sof/drivers/sai.h]: Fair. Still, in the future better explanations are warranted. @dbaluta Is this ready to merge?

**Comment by dbaluta** [comment]
@groncarolo Looks good now! thanks!

**Pull request merged**